controllers/cart.js

```diff
@@ -0,0 +1,63 @@
+const { products, carts, orders } = require('../storage');
+const crypto = require('crypto');
+
+function renderCart(id) {
+  if (carts.find((item) => item.userId === id)) {
+    return;
+  } else {
+    const cart = {
+      id: crypto.randomUUID(),
+      userId: id,
+      products: [],
+    };
+    carts.push(cart);
+  }
+}
+
+function addProductToCart(id, product) {
+  const thisProduct = products.find((item) => item.id === +product);
+  const thisCart = carts.find((item) => item.userId === id);
+  const arrOfProducts = thisCart.products;
+  arrOfProducts.push(thisProduct);
+  thisCart.products = arrOfProducts;
```

Comment:
you don't need to reassign an array to a new variable, you can push directly to the products array:
`thisCart.products.push(thisProduct)`